src/Nest/XPack/MachineLearning/Job/Config/MachineLearningFilter.cs

```diff
@@ -0,0 +1,19 @@
+﻿using System.Collections.Generic;
+using Newtonsoft.Json;
+
+namespace Nest
+{
+	[JsonObject]
+	public class MachineLearningFilter
+	{
+		// TODO: Looks like type is only included when INCLUDE_TYPE_KEY set. Determine if this needs to be exposed
```

Comment:
@russcam ?

Reply:
I'd err on not exposing this right now. It's an internal implementation detail (/cc @droberts195  could you confirm?)

Reply:
ML filters are currently an undocumented feature of ML that hasn't been properly tested and might change before it's officially released.  So it shouldn't be documented for the NEST client either.

Even when ML filters is eventually documented, `INCLUDE_TYPE_KEY` shouldn't be present in any client code.  You're correct that it's an internal implementation detail.  It's a `toXContent` parameter, i.e. a quirk of the Java code, not an API parameter.

(For interest, the reason it exists is that ML stores the filters in an Elasticsearch index, and when filters are serialised to this index they need an extra field so that we can distinguish them from other documents stored in the same index.  But when they're returned via the ML REST endpoints we take out this extra field so the user sees just the filter they originally created.  This is all implementation details though.  The end user doesn't need to know how filters are stored.)

Reply:
@codebrain Based on this, I think we should remove the `GetFilters` API altogether

Reply:
@russcam - Get/Put/Delete filters, all REST spec definitions and all associated tests?

Reply:
Deleted!

Reply:
Thinking we should pull them out into a separate branch; when this branch is merged, the commits will be squashed so whilst the commits will be reachable in the reflog for some time, I think it'll be easier to get at these if we park them in a separate branch